### PR TITLE
chore(render): add MultiRenderer

### PR DESCRIPTION
**Pull request by efekarakus**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/term/progress/render.go [hunk 1]
```diff
@@ -78,3 +78,44 @@ func EraseAndRender(out FileWriteFlusher, r Renderer, prevNumLines int) (int, er
...
 }
+
+// MultiRenderer returns a Renderer that's the concatenation of the input renderers.
+// The renderers are rendered sequentially, and the MultiRenderer is only Done once all renderers are Done.
+func MultiRenderer(renderers ...DynamicRenderer) DynamicRenderer {
+	mr := &multiRenderer{
+		renderers: renderers,
+		done:      make(chan struct{}),
+	}
+	go mr.listen()
```
Comment (dannyrandall) [inline internal/pkg/term/progress/render.go:89]: hahah go mr. listen! 🥳

**Review by dannyrandall — APPROVED** [review]

File: internal/pkg/term/progress/render.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/term/progress/render.go:89]: I chuckled at that too

File: internal/pkg/term/progress/component.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ type noopComponent struct{}
 
 // Render does not do anything.
 // It returns 0 and nil for the error.
-func (c *noopComponent) Render(out io.Writer) (numLines int, err error) {
+func (c *noopComponent) Render(_ io.Writer) (numLines int, err error) {
```
Comment (iamhopaul123) [inline internal/pkg/term/progress/component.go:23]: Why not just
```suggestion
func (c *noopComponent) Render(io.Writer) (numLines int, err error) {
```

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/term/progress/component.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/term/progress/component.go:23]: oh no reason, I didn't think of it 😂

**Pull request merged**